### PR TITLE
py-numpy: add v1.26.4

### PR DESCRIPTION
**Pull request by adamjstewart**
https://github.com/numpy/numpy/releases/tag/v1.26.4

Test results:
```
=========================== short test summary info ============================
FAILED ../linalg/tests/test_linalg.py::TestDet::test_types[complex64] - Runti...
FAILED ../linalg/tests/test_linalg.py::TestDet::test_types[complex128] - Runt...
FAILED ../tests/test_public_api.py::test_api_importable - UserWarning: Distut...
= 3 failed, 34998 passed, 380 skipped, 45 xfailed, 2 xpassed in 186.31s (0:03:06) =
```
First two failures are `RuntimeWarning: divide by zero encountered in det`. Last failure is `UserWarning: Distutils was imported before Setuptools`.

~Depends on #41604~

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-numpy/package.py
Comment (rgommers) [inline var/spack/repos/builtin/packages/py-numpy/package.py]: minor: this works, but should it be `0.15:`? We'll be releasing 0.16.0 very soon.

**Review by alalazo — DISMISSED** [review]
Feel free to merge when you're ready, if you're not waiting for further reviews

**Review by rgommers — APPROVED** [review]
Just realized I missed this 3 weeks ago, apologies. LGTM, just one minor comment (happy either way on that suggestion, merging as is works too).

File: var/spack/repos/builtin/packages/py-numpy/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-numpy/package.py]: This hearkens back to our scipy debate over whether we should relax all dependencies until we discover incompatibilities or if we should strictly follow the dependencies declared in pyproject.toml. I'll let you make the final call and document whatever we decide.

Comment (rgommers) [reply var/spack/repos/builtin/packages/py-numpy/package.py]: Ah yes, I see I didn't respond again to that on gh-42213.  Let me do that now.

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Review by alalazo — APPROVED** [review]

**Pull request merged**
